libs/api/domains/auth-admin/src/lib/application/applications.service.ts

```diff
@@ -23,27 +24,52 @@ export class ApplicationsService {
     return resp
   }
 
-  createApplication(input: CreateApplicationInput) {
-    const displayName = new TranslatedValue()
-    displayName.locale = 'is'
-    displayName.value = input.displayName
+  async createApplication(
+    input: CreateClientsInput,
+    user: Auth,
+  ): Promise<CreateApplicationResponseDto[]> {
+    const applicationRequest = {
+      tenantId: input.tenantId,
+      adminCreateClientDto: {
+        clientId: input.applicationId,
+        clientType: (input.applicationType as string) as AdminCreateClientDtoClientTypeEnum,
+        tenantId: input.tenantId,
+        nationalId: user.nationalId,
+        clientName: input.displayName,
+      },
+    }
 
-    const application = new Application()
-    application.applicationId = input.applicationId
-    application.applicationType = input.applicationType
-    application.environments = input.environments.map((env) => {
-      const applicationEnv = new ApplicationEnvironment()
-      applicationEnv.name = input.applicationId
-      applicationEnv.environment = env
-      applicationEnv.displayName = [displayName]
-      applicationEnv.ApplicationUrls.callbackUrls = []
+    const createApplicationResponse = [] as CreateApplicationResponseDto[]
 
-      return applicationEnv
-    })
+    const created = await Promise.all(
```

Comment:
This will reject if there is an error in one of the environment. We should use `Promise.allSettled` instead which will return an object with the promise status and the value. Then we can check the status and map the response accordingly.